iOS/DuckDuckGo/Subscription/FreeTrials/VPNSubscriptionPromotionHelper.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  VPNSubscriptionPromotionHelper.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import BrowserServicesKit
+import Core
+import Foundation
+import Subscription
+
+/// Protocol defining the interface for the VPN Subscription promotion helper.
+///
+/// Conforming types provide logic for determining how the Subscription should be promoted on the VPN menu item,
+/// as well as utilities for persistence and pixel firing related to the promotion.
+protocol VPNSubscriptionPromotionHelping {
+
+    /// Status of the subscription promotion.
+    var subscriptionPromoStatus: VPNSubscriptionPromotionStatus { get }
+
+    /// Provides the URL components for subscribing as part of the promotion.
+    func subscriptionURLComponents() -> URLComponents?
+
+    /// Records when the promo has been shown to the user.
+    func subscriptionPromoWasShown()
+
+    /// Fires a pixel when the network protection promotion is tapped by the user.
+    func fireTapPixel()
+}
+
+enum VPNSubscriptionPromotionStatus {
+    case promo
+    case noPromo
+    case subscribed
+
+    var pixelParameter: String {
+        switch self {
+        case .promo:
+            return "pill"
+        case .noPromo:
+            return "no_pill"
+        case .subscribed:
+            return "subscribed"
```

Comment:
Despite the UI changes from the kickoff discussion (showing a dot instead of a pill), I left these pixel parameters as-is since they have already gone through pixel triage. Raised here in case we want to change it before shipping: https://app.asana.com/1/137249556945/task/1211552057851668/comment/1211710076965759?focus=true